### PR TITLE
[DO NOT MERGE] Upgrade webpack

### PR DESCRIPTION
**Pull request by mparke**
Upgrades to webpack 2.0:
https://webpack.js.org/configuration/

Uses the webpack dev server + hot reloader:
https://webpack.js.org/configuration/dev-server/

Uses updated react plugin for hot reloading and splits `index.js` into `index.js` and `app.js` to work with reloader as per docs:
https://webpack.js.org/guides/hmr-react/

Changes our default transform preset from `0` to `3`, please read details here: https://babeljs.io/docs/plugins/#presets-stage-x-experimental-presets-

### CODE REVIEW

**Comment by mparke** [comment]
Needs:

- [ ] Shrinkwrap Update
- [ ] Fixed tests
- [ ] Cleaned up comments
- [ ] Checked environment vars

**Comment by eatonphil** [comment]
Lish is not working in this PR and for some reason the settings page of a Linode is not working either... But that may have been an issue I created?

**Comment by mparke** [comment]
Fixed several pages that broke from bad imports @eatonphil

**Comment by mparke** [comment]
Changing back to DO NOT MERGE for now, until further testing can be done

File: src/linodes/linode/settings/layouts/index.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-export IndexPage from './IndexPage';
-export AlertsPage from './AlertsPage';
-export DisplayPage from './DisplayPage';
+export { IndexPage } from './IndexPage';
+export { AlertsPage } from './AlertsPage';
+export { DisplayPage } from './DisplayPage';
```
Comment (eatonphil) [inline src/linodes/linode/settings/layouts/index.js]: Why do these need to change?

**Comment by mparke** [comment]
Review and note any features that utilize `< stage-3` presets

https://github.com/tc39/proposal-class-public-fields
https://github.com/tc39/proposal-object-rest-spread

**Comment by mparke** [comment]
closing after https://github.com/linode/manager/pull/2278, can be referenced as needed

**Pull request closed without merging**